python_bindings/tutorial/lesson_12_using_the_gpu.py

```diff
@@ -255,72 +226,85 @@ def test_correctness(self, reference_output):
         assert output.type() == hl.UInt(8)
 
         # Check against the reference output.
-        for c in range(self.input.channels()):
-            for y in range(self.input.height()):
-                for x in range(self.input.width()):
-                    if output[x, y, c] != reference_output[x, y, c]:
-                        print(
-                            "Mismatch between output (%d) and "
-                            "reference output (%d) at %d, %d, %d" % (
-                                output[x, y, c],
-                                reference_output[x, y, c],
-                                x, y, c))
-                        return
+        for cc in range(self.input.channels()):
+            for yy in range(self.input.height()):
+                for xx in range(self.input.width()):
+                    assert output[xx, yy, cc] == reference_output[xx, yy, cc], \
+                        "Mismatch between output (%d) and reference output (%d) at %d, %d, %d" % (
+                        output[xx, yy, cc],
+                        reference_output[xx, yy, cc],
+                        xx, yy, cc)
 
         print("CPU and GPU outputs are consistent.")
 
 
 def main():
     # Load an input image.
-    image_path = os.path.join(os.path.dirname(__file__), "../../tutorial/images/rgb.png")
+    image_path = os.path.join(os.path.dirname(
+        __file__), "../../tutorial/images/rgb.png")
     input = hl.Buffer(imageio.imread(image_path))
```

Comment:
The Buffer constructor that takes whatever imread returns should probably set host dirty.